lib/Zonemaster/Backend/DB/SQLite.pm

```diff
@@ -164,97 +165,124 @@ sub create_new_batch_job {
+        # A recent entry exists, so return its id
+        if ( $recent_id > $self->config->force_hash_id_use_in_API_starting_from_id() ) {
+            $result_id = $recent_hash_id;
+        }
+        else {
+            $result_id = $recent_id;
+        }
+    }
+    else {
+        $dbh->do(
+            q[INSERT INTO test_results (batch_id, priority, queue, params_deterministic_hash, params, domain, test_start_time, undelegated) VALUES (?, ?,?,?,?,?, datetime('now'),?)],
+            undef,
+            $batch_id,
+            $priority,
+            $queue,
+            $test_params_deterministic_hash,
+            $encoded_params,
+            $test_params->{domain},
+            ($test_params->{nameservers})?(1):(0),
+        );
+        
+        $dbh->do(q[UPDATE test_results SET hash_id = ? WHERE params_deterministic_hash = ?], undef, substr(md5_hex(time().rand()), 0, 16), $test_params_deterministic_hash);
+        
+        my ( $id, $hash_id ) = $dbh->selectrow_array(
+            "SELECT id, hash_id FROM test_results WHERE params_deterministic_hash='$test_params_deterministic_hash' ORDER BY id DESC LIMIT 1" );
+            
+        if ( $id > $self->config->force_hash_id_use_in_API_starting_from_id() ) {
+            $result_id = $hash_id;
+        }
+        else {
+            $result_id = $id;
+        }
+    }
 
-    return $result;
+    return $result_id;
 }
 
 sub test_progress {
     my ( $self, $test_id, $progress ) = @_;
 
-    $self->dbh->do( "UPDATE test_results SET progress=$progress WHERE id=$test_id AND progress <> 100" ) if ( $progress );
+    my $id_field = $self->_get_allowed_id_field_name($test_id);
+
+    my $dbh = $self->dbh;
+    if ( $progress ) {
+        if ($progress == 1) {
+            $dbh->do( "UPDATE test_results SET progress=?, test_start_time=datetime('now') WHERE $id_field=? AND progress <> 100", undef, $progress, $test_id );
+        }
+        else {
+            $dbh->do( "UPDATE test_results SET progress=? WHERE $id_field=? AND progress <> 100", undef, $progress, $test_id );
+        }
+    }
 
-    my ( $result ) = $self->dbh->selectrow_array( "SELECT progress FROM test_results WHERE id=$test_id" );
+    my ( $result ) = $self->dbh->selectrow_array( "SELECT progress FROM test_results WHERE $id_field=?", undef, $test_id );
 
     return $result;
 }
 
 sub get_test_params {
     my ( $self, $test_id ) = @_;
 
-    my $result;
+    my $id_field = $self->_get_allowed_id_field_name($test_id);
+    my ( $params_json ) = $self->dbh->selectrow_array( "SELECT params FROM test_results WHERE $id_field=?", undef, $test_id );
 
-    my ( $params_json ) = $self->dbh->selectrow_array( "SELECT params FROM test_results WHERE id=$test_id" );
-    eval { $result = decode_json( $params_json ); };
-    die "$@ \n" if $@;
+    my $result;
+    eval {
+        $result = decode_json( $params_json );
+    };
+    
+    warn "decoding of params_json failed (testi_id: [$test_id]):".Dumper($params_json) if $@;
```

Comment:
`test_id` is misspelled.

Also, if you change `warn` to use `Log::Any::warn()` instead, the log message gets prepended with a nice header.